### DISCUSSION TITLE
How do I recognize, which order I just got a receipt for? [iOS]

### DISCUSSION THREAD

**Discussion by 5ZYSZ3K** [hyochan/react-native-iap#2111] [Q&A]
## I have the following problem:

When a  called with a receipt, how do I tell, which order is this receipt for? I can send the transaction object to my backend, when all the order data is stored, but how do I recognize, which order got just paid for?

- In my app, user can order a product with some addons. For the product they will pay using in-app purchases (since it is a "digital good"), and to pay for addons, they will need to use an external site (since addons are physical items). The user can also buy this product as many times as they want, and all those orders may differ from each other (because of addons, consents and other things). So, when the purchaseUpdateListener gets called, how do I tell, which order has just been paid for?

- The purchaseUpdateListener callback may get called at any time, even when the user has already logged out, and all the order data got removed from my application's state. How may i tell then, what just got paid for?

I've found a "transactionId" field that I receive with a transaction receipt, but I couldn't found this field earlier on the payment process. Can i get this field's value just after the user requests a purchase, so that I can send it with the current order's id for the backend to recognize the receipt later?

I've also tried recognizing it by using "appAccountToken" field, but to no avail. I couldn't get this token on my backend. Tried [this solution](https://github.com/dooboolab/react-native-iap/discussions/2034), but it does not seem to be working in my case

**Comment by andresesfm** [comment]
`productId`?
